test/integration/submission/submission-schema-migration.spec.ts

```diff
@@ -229,85 +244,293 @@ describe('schema migration api', () => {
 
-      chai.assert(sendProgramUpdatedMessageFunc.calledOnceWith(programId));
+      await migrateSyncTo('2.0').then(async (res: any) => {
+        res.should.have.status(200);
+        const updatedDonor = await findInDb(dburl, 'donors', {});
+
+        // donor 1 stats after migraiton
+        chai.expect(updatedDonor[0].aggregatedInfoStats).to.deep.include({
+          submittedCoreFields: 2,
+          expectedCoreFields: 2,
+        });
+        // donor 2 stats after migraiton
+        chai.expect(updatedDonor[1].aggregatedInfoStats).to.deep.include({
+          submittedCoreFields: 5,
+          expectedCoreFields: 10,
+        });
+        // this stub schema has also turned an existing required field to core in primary diagnosis
+        chai.expect(updatedDonor[1].primaryDiagnosis.clinicalInfoStats).to.deep.include({
+          submittedCoreFields: 3,
+          expectedCoreFields: 8,
+        });
+        // donor 3 stats after migraiton, now has recently calculated stats, including the one without any stats
+        chai.expect(updatedDonor[2].aggregatedInfoStats).to.deep.include({
+          submittedCoreFields: 2,
+          expectedCoreFields: 2,
+        });
+
+        chai.assert(sendProgramUpdatedMessageFunc.calledOnceWith(programId));
+      });
+    });
+    it('should update the schema after an enum option was removed, and make donor2 invalid', async () => {
+      await migrateSyncTo('7.0').then(async (res: any) => {
+        res.should.have.status(200);
+        const schema = (await findInDb(dburl, 'dataschemas', {})) as SchemasDictionary[];
+        schema[0].version.should.eq('7.0');
+      });
+
+      const res = await getAllMigrationDocs();
+
+      const [migration] = res.body;
+      migration.toVersion.should.equal('7.0');
+
+      migration.stage.should.not.equal(migratationFailStage);
+      migration.should.not.have.property('newSchemaErrors');
+      migration.invalidDonorsErrors.length.should.equal(1);
+      const errorObj = migration.invalidDonorsErrors[0].errors[0].primary_diagnosis[0];
+      chai
+        .expect(errorObj)
+        .to.have.property('errorType', SchemaValidationErrorTypes.INVALID_ENUM_VALUE);
+      chai.expect(errorObj).to.have.property('fieldName', TUMOUR_STAGING_SYSTEM);
+    });
+    it('should update the schema after a new required field is added, and make all donors invalid', async () => {
+      await migrateSyncTo('8.0').then(async (res: any) => {
+        res.should.have.status(200);
+        const schema = (await findInDb(dburl, 'dataschemas', {})) as SchemasDictionary[];
+        schema[0].version.should.eq('8.0');
+      });
+
+      const res = await getAllMigrationDocs();
+      const donors = await findInDb(dburl, 'donors', {});
+
+      const [migration] = res.body;
+      migration.toVersion.should.equal('8.0');
+
+      migration.stage.should.not.equal('FAILED');
+      migration.should.not.have.property('newSchemaErrors');
```

Comment:
might want to make this a const, its used a lot in this file. also isn't their a const for `FAILED`

Reply:
good catch! I declared a const for it but I missed these!